cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -272,21 +267,14 @@ public ApplicationMeta getApplication(String namespaceId, String appId, String v
    */
   public Map<ApplicationId, ApplicationMeta> getApplicationsForAppIds(Collection<ApplicationId> appIds)
     throws IOException {
-    if (appIds.isEmpty()) {
```

Comment:
is this removed because the incoming collection can never be empty?

Reply:
Because the Storage SPI should return an empty map on empty keys.